packages/firestore/src/core/types.ts

```diff
@@ -39,31 +39,32 @@ export type MutationBatchState = 'pending' | 'acknowledged' | 'rejected';
  * offline (e.g. get() calls shouldn't wait for data from the server and
  * snapshot events should set metadata.isFromCache=true).
  */
-export type OnlineState =
+export const enum OnlineState {
   /**
    * The Firestore client is in an unknown online state. This means the client
    * is either not actively trying to establish a connection or it is currently
    * trying to establish a connection, but it has not succeeded or failed yet.
    * Higher-level components should not operate in offline mode.
    */
-  | 'Unknown'
+  Unknown = 'Unknown',
```

Comment:
Do we need the strings in here anymore? Can these just be numeric?

Reply:
The string values are persisted in WebStorage (but it is mostly transient data). I added a comment to explain this.